src/forms/inputs/file-input/file-input.js

```diff
@@ -121,11 +124,12 @@ class FileInput extends React.Component {
 }
 
 // eslint-disable-next-line react/prop-types
-function renderPreview ({ file, value, thumbnail, hidePreview, previewComponent: Component, children }) {
-  if (hidePreview) return null
+function renderPreview ({ file, value, thumbnail, previewComponent: Component, children }) {
   if (Component) return <Component file={ file } />
   if (children) return children
-  return <ImagePreview image={ value || thumbnail } />
+  const renderImagePreview = isImageType(file) || thumbnail
+  if (renderImagePreview) return <ImagePreview image={ value || thumbnail } />
+  return <FilePreview file={ file } />
```

Comment:
Consider putting the file null check to simplify the file preview component:
`return file ? <FilePreview file={ file } /> : null`

It's good how it is, just an optional idea.

Reply:
Codeclimate got mad at me for cognitive complexity of that function if I added one more ternary 🙄

Reply:
I always just mark those as `wontfix` 😀